appinventor/buildserver/src/com/google/appinventor/buildserver/resources/runtime.scm

```diff
@@ -407,7 +408,8 @@
            ;; for event handlers and component names.
            (let ((registeredObject (string->symbol registeredComponentName)))
                  (if (is-bound-in-form-environment registeredObject)
-                     (if (eq? (lookup-in-form-environment registeredObject) componentObject)
+                     ;; temporarily deleted because I have no ideas how to get true out of this condition using isMulti
+                     ;;(if (or (eq? (lookup-in-form-environment registeredObject) componentObject) (isMulti))
```

Comment:
On the surface, the line of code you mention does't work because you have parentheses around 'isMulti'.  In Scheme those parentheses cause a procedure call, so '(isMulti)' means call the procedure that is currently bound to isMulti, and since isMulti will actually be bound to a boolean it fails.

I should point out, though, that your rationale for why it's ok to avoid the lookup isn't really correct, since you're not really taking into account the purpose of the check (i.e., the whole renaming issue).  Perhaps rather than registering the handler to an arbitrary instance (which is kinda hacky anyway and what happens when that component gets deleted) you can extend the dispatch mechanism to support registering event handlers to component types in addition to component instances.